File: leetcode/swift/transpose-matrix/0898-transpose-matrix/0898-transpose-matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        rows = len(matrix[0])
        cols = len(matrix)
        res = [ ]

        for r in range(rows):
            newR = []
            for c in range(cols):
                newR.append(matrix[c][r])
            res.append(newR)

        return res

File: leetcode/swift/transpose-matrix/0898-transpose-matrix/0898-transpose-matrix.swift
class Solution {
    func transpose(_ matrix: [[Int]]) -> [[Int]] {
        let rows = matrix.count
        let cols = matrix[0].count
        var mat = [[Int]](repeating: Array(repeating: 0, count: rows), count: cols)

        for r in 0..<rows {
            for c in 0..<cols {
                mat[c][r] = matrix[r][c]
            }
        }

        return mat
    }
}